server/src/domain/types.rs

```diff
@@ -167,6 +167,82 @@ impl ValueType for UserId {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(from = "String")]
+pub struct DisplayName(String);
+
+impl DisplayName {
+    pub fn new(display_name: &str) -> Self {
+        Self(display_name.to_string())
+    }
+
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+
+    pub fn into_string(self) -> String {
+        self.0
+    }
+}
+
+impl std::fmt::Display for DisplayName {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<String> for DisplayName {
+    fn from(s: String) -> Self {
+        Self::new(&s)
+    }
+}
+
+impl From<DisplayName> for Value {
+    fn from(display_name: DisplayName) -> Self {
+        display_name.into_string().into()
+    }
+}
+
+impl From<&DisplayName> for Value {
+    fn from(display_name: &DisplayName) -> Self {
+        display_name.as_str().into()
+    }
+}
+
+impl TryGetable for DisplayName {
+    fn try_get(res: &QueryResult, pre: &str, col: &str) -> Result<Self, TryGetError> {
+        Ok(DisplayName::new(&String::try_get(res, pre, col)?))
```

Comment:
Same here, you're making an extra copy. Since you have a `From<String>`, you can write
`Ok(DisplayName::from(String::try_get(res, pre, col)?))`